src/protocols/hdWallet.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  derivePathFromKey,
+  derivePathFromSeed,
+  getKeyPair,
+} from '@aeternity/aepp-sdk';
+import type {
+  IWallet,
+  Protocol,
+} from '@/types';
+import {
+  PROTOCOL_AETERNITY,
+  PROTOCOL_BITCOIN,
+} from '@/constants';
+import { AeternityProtocol } from './aeternity/libs/AeternityProtocol';
+import { BitcoinProtocol } from './bitcoin/libs/BitcoinProtocol';
+
+export function generateHdWallet(seed: Uint8Array, protocol: Protocol) {
```

Comment:
I think this should be moved to each of the protocol adapters.

Reply:
Exactly, That worked better. Please have a look @peronczyk